src/get-started/java-binding.md

```diff
@@ -401,39 +377,66 @@ An exception of any other type will be recorded with _no_ error code
 as an "unexpected" execution error.
 
 If transaction execution fails, the changes made by the transaction are
-rolled back, while the error data is stored in the database for further user
-reference. Light clients also provide access to information on the
-transaction execution result (which may be either success or failure)
+rolled back, while the error data is [stored in the database][call-errors-registry]
+for further user reference. Light clients also provide access to information
+on the transaction execution result (which may be either success or failure)
 to their users.
 
-#### After Transactions Handler
+[execution-exception]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/core/transaction/ExecutionException.html
+[call-errors-registry]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/core/blockchain/Blockchain.html#getCallErrors(long)
 
-Services can receive a notification after all transactions in a block has been
-processed but before the block has been committed, allowing inspecting
-the changes made by the transactions and modifying any service state.
+#### Submitting Transactions to Self
 
-Exonum delivers this notification to implementations
-of [`Service#beforeCommit`][service-before-commit] method. The method will see
-the state of the database after all transactions have been applied.
-It may make any changes to the service state, which will be included
+If the service itself needs to create a transaction on a particular node,
+it can use the [`Node#submitTransaction`][node-submit-transaction] method.
+This method will create and sign a transaction message using
+the [service key][node-configuration-validator-keys]
+of _that particular node_ (meaning that the node will be the author
+of the transaction), and submit it to the network.
+Invoking this method on each node unconditionally will produce
+_N_ transactions that have the same payloads, but different
+authors’ public keys and signatures, where _N_ is the number of nodes
+in the network.
+
+[node-configuration-validator-keys]: ../architecture/configuration.md#validator-keys
+
+#### Before and After Transactions Handlers
+
+Services can receive notifications:
+
+- before any transactions in a block has been processed
+- after all transactions in a block has been processed, but before the block
+  has been committed.
+
+Such notification allow inspecting the changes made by the transactions
+and modifying any service state.
```

Comment:
IMO, "any service state" implies "state of any service", which is incorrect - the service may (or at least should) only modify its own state.